Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Comments/CQRS/Handlers/UpdateCommentCommandHandler.cs

```diff
@@ -0,0 +1,62 @@
+ 
+using AutoMapper;
+using BlogApp.Application.Contracts.Persistence;
+using BlogApp.Application.Exceptions;
+using BlogApp.Application.Features.Blog.DTOs;
+using BlogApp.Application.Features.Comments.CQRS.Commands;
+using BlogApp.Application.Features.Comments.DTOs;
+using BlogApp.Application.Features.Comments.DTOs.Validators;
+using BlogApp.Application.Responses;
+using MediatR;
+
+namespace BlogApp.Application.Features.Comments.CQRS.Handlers;
+
+public class UpdateCommentCommandHandler : IRequestHandler<UpdateCommentCommand, Result<UpdateCommentDto>>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IMapper _mapper;
+
+    public   UpdateCommentCommandHandler(IUnitOfWork unitOfWork,IMapper mapper)
+    {
+        _unitOfWork = unitOfWork;
+        _mapper = mapper;
+        
+    }
+
+    public async Task<Result<UpdateCommentDto>> Handle(UpdateCommentCommand request, CancellationToken cancellationToken)
+    {
+
+        var response = new Result<UpdateCommentDto?>();
+        var validator = new UpdateCommentDtoValidator();
+        var validationResult = await validator.ValidateAsync(request.CommentDto);
+ 
+        if (validationResult.IsValid == true){
+            var comment = await _unitOfWork._CommentRepository.Get(request.CommentDto.Id);
+            _mapper.Map(request.CommentDto, comment);
+
+            await _unitOfWork._CommentRepository.Update(comment);
```

Comment:
You should check if the comment exists before updating it

Reply:
I think I didn't include the exsistence validation for update comment Dto ,  now I already fixed it  by editing the validation .
Also I have updated the dto only to edit the content.